tests/Integration/inc/Engine/Optimization/GoogleFonts/Subscriber/preconnect.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\Optimization\GoogleFonts\Subscriber;
+
+use WPMedia\PHPUnit\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Optimization\GoogleFonts\Subscriber::preconnect
+ * @group CombineGoogleFonts
+ */
+class Test_Preconnect extends TestCase {
+    /**
+	 * @dataProvider providerTestData
+	 */
+	public function testShouldReturnExpectedArray( $option_value, $urls, $relation_type, $expected ) {
+        $callback = function() use ( $option_value ) {
+            return $option_value;
+        };
+
+		add_filter( 'pre_get_rocket_option_minify_google_fonts', $callback );
+
+		$this->assertSame(
+			$expected,
+			apply_filters( 'wp_resource_hints', $urls, $relation_type )
+        );
+
+        remove_filter( 'pre_get_rocket_option_minify_google_fonts', $callback );
```

Comment:
@Tabrisrp Instead of using a closure as a callback, let's make it a method and then move the `remove_filter()` to the `tearDown()` method. 

Why? If the test fails, the `remove_filter()` will not run. But by moving it into the `tearDown()`, we make sure that it's unhooked after each test.

Reply:
how can I pass the value from `$option_value` to the method in that case?

Reply:
Set it to a private property in the test class. Then reset that property in `setUp()`.

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/2524/commits/befeef31c3b583bc0ef34ee67cf2c504a7e7cb56